seqr/views/utils/test_utils.py

```diff
@@ -1,25 +1,90 @@
 # Utilities used for unit and integration tests.
-from django.contrib.auth.models import User
+from django.contrib.auth.models import User, Group
+from django.test import TestCase
+from guardian.shortcuts import assign_perm
+import json
+from seqr.models import Project, CAN_VIEW, CAN_EDIT
 
-def _check_login(test_case, url):
-    """For integration tests of django views that can only be accessed by a logged-in user,
-    the 1st step is to authenticate. This function checks that the given url redirects requests
-    if the user isn't logged-in, and then authenticates a test user.
 
-    Args:
-        test_case (object): the django.TestCase or unittest.TestCase object
-        url (string): The url of the django view being tested.
-     """
-    response = test_case.client.get(url)
-    test_case.assertEqual(response.status_code, 302)  # check that it redirects if you don't login
+class AuthenticationTestCase(TestCase):
 
-    test_user = User.objects.get(username='test_user')
-    test_case.client.force_login(test_user)
+    STAFF = 'staff'
+    MANAGER = 'manager'
+    COLLABORATOR = 'collaborator'
+    ANY = 'any'
 
+    @classmethod
+    def setUpTestData(cls):
+        cls.staff_user = User.objects.get(username='test_user')
+        cls.manager_user = User.objects.get(username='test_user_manager')
+        cls.collaborator_user = User.objects.get(username='test_user_non_staff')
+        cls.no_access_user = User.objects.get(username='test_user_no_access')
 
-def login_non_staff_user(test_case):
-    test_user = User.objects.get(username='test_user_non_staff')
-    test_case.client.force_login(test_user)
+        edit_group = Group.objects.get(pk=2)
+        view_group = Group.objects.get(pk=3)
+        edit_group.user_set.add(cls.manager_user)
+        view_group.user_set.add(cls.manager_user, cls.collaborator_user)
+        assign_perm(user_or_group=edit_group, perm=CAN_EDIT, obj=Project.objects.all())
+        assign_perm(user_or_group=view_group, perm=CAN_VIEW, obj=Project.objects.all())
```

Comment:
Due to limitations on how django loads built in permissions and groups, its impossible to create these permissions in the fixtures so they need to be created here at runtime